### DISCUSSION TITLE
Hardware requirements

### DISCUSSION THREAD

**Discussion by dzdidi** [ACINQ/eclair#2891] [Q&A] [answered]
I've seen this benchmark being referenced in one of replies https://github.com/bottlepay/lightning-benchmark and that your team had not tried running eclair on raspberry pi. 

Could you please share the minimum and recommended hardware specifications required to operate Eclair effectively? Additionally, if available, I would greatly appreciate any performance metrics related to the number of channels opened, payments processed, and forwarded transactions.

Such information would be invaluable for estimating the costs associated with deploying Eclair as a routing or LSP node.

Thanks

**Comment by t-bast** [comment]
> Could you please share the minimum and recommended hardware specifications required to operate Eclair effectively? 

There is no good answer to this: hardware requirements directly depend on the number of active channels you expect to have and number of payments you expect to relay, as resource usage scales based on those metrics. You should probably ask other node operators what they use and the size of their node to get a rough idea of what to use (e.g. on the t.me/eclair_lightning telegram group).

Note that eclair is mostly RAM-intensive, and you should probably tweak the default JVM configuration as detailed [here](https://github.com/ACINQ/eclair?tab=readme-ov-file#java-environment-variables).

You should also carefully read the [configuration file](https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/resources/reference.conf) and tweak what makes sense for your node.

> I would greatly appreciate any performance metrics related to the number of channels opened, payments processed, and forwarded transactions.

Such performance metrics don't make sense at all today: the payment volume is low enough that there are simply no performance issues related to payments. The only thing that's important is to scale your machine based on the number of active channels you expect to have.

**Reply by dzdidi** [reply]
@t-bast thank you for references and quick reply.

Completely agree that number of active channels is the most important factor.

Do you perhaps have some rough estimate on how many active channels Eclair node on, let's say 4vCPU, 2 Core, 16GB RAM machine, can handle? 

Or if you have some "magic number"  like `X active channels per 1 GB of RAM` I'll gladly accept this as an answer too.

**Reply by pm47** [reply, marked as answer]
> some rough estimate on how many active channels Eclair node on, let's say 4vCPU, 2 Core, 16GB RAM machine, can handle?

With this hardware you should be able to comfortably handle 25,000 channels. Setting `-Xmx 12G` should be a good starting point. For better security, and to handle a large number of concurrent connections, consider the [cluster mode](https://github.com/ACINQ/eclair/blob/master/docs/Cluster.md).

**Reply by dzdidi** [reply]
Thank you

**Comment by DerEwige** [comment]
> Do you perhaps have some rough estimate on how many active channels Eclair node on, let's say 4vCPU, 2 Core, 16GB RAM machine, can handle?

I run my eclair on this setup:

Application Server:
4vCPU AMD EPYC 7R13, 16 GB Ram
Eclair, Bitcoin Core and some other applications

Primary DB Server:
2vCPU Graviton2, 4 GB Ram
Primary Postgres Server (synchronous replication to secondary)

Secondary DB Server:
2vCPU Graviton2, 1 GB Ram

I have only 50-60 channels, but at my peak I did 8 payments per second over several days. (rebalancing)
So about 700'000 payments per day.
Maximum Channel DB size reached was 250 million entries. (About 200 GB DB size)

Probably not your use case. But I think it nicely showcases, what kind of throughput that kind of setup can bring.
Maybe that info is usefull to you.
